sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1666,6 +1684,82 @@ def get_custom_materialization_type(name: str) -> t.Type[CustomMaterialization]:
     return strategy_type
 
 
+class EngineManagedStrategy(MaterializableStrategy):
+    def create(
+        self,
+        table_name: str,
+        model: Model,
+        is_table_deployable: bool,
+        is_snapshot_deployable: bool,
+        **render_kwargs: t.Any,
+    ) -> None:
+        if is_table_deployable and is_snapshot_deployable:
+            # We could deploy this to prod; create a proper managed table
```

Comment:
I've now realized that this should actually work nicely. Even if the snapshot is not deployable in dev, next time the change is applied to prod, SQLMesh should detect a missing table and try to run this logic again, at which point the snapshot should already be deployable.

Reply:
No, that's actually not true. I think the snapshot will still be non-deployable the first time it's promoted to prod. I think we're still missing a story around creating a managed table when a non-deployable snapshot gets promoted to prod.

Reply:
> I think the snapshot will still be non-deployable the first time it's promoted to prod

I tested this manually and it seemed to work (since `is_snapshot_deployable` doesnt come from the doctored DeployabilityIndex that gets passed to `create()`, it comes from the actual DeployabilityIndex).

However, of course a manual test wont verify the behaviour in code or prevent regressions, so i'll see how I can add an automated test for this scenario